vim/core/oni-core-interop/plugin/init.vim

```diff
@@ -42,6 +52,72 @@ function OniNotifyEvent(eventName)
     call OniNotify(["event", a:eventName, context])
 endfunction
 
+function User_buffers() " help buffers are always unlisted, but quickfix buffers are not
+  return filter(range(1,bufnr('$')),'buflisted(v:val) && "quickfix" !=? getbufvar(v:val, "&buftype")')
+endfunction
+
+function OniGetAllBuffers()
+  let l:buffers = []
+  let l:bufnums = User_buffers()
+  if exists("l:bufnums")
+    for l:bufnum in l:bufnums
+      try
+        let l:buffer = OniGetEachContext(l:bufnum)
+        let l:buffers += [l:buffer]
+      catch /.*/
+      "Probably dont want this outside of a debugging scenario
```

Comment:
Interesting - are there any cases we could hit this?

Reply:
Not really, nor that i've observed with the current state of the function I have it there as a last ditch attempt at catching an error if one occurs but it was essentially there for me to use to debug whilst writing it but also the try/catch would mean that even if there were an error the user wouldn't be faced with vim's not so pleasant scrolling wall of red lines

i've checked that each function gets the right kind of argument and that it doesn't recieve an invalid argument which in most cases would return in 0 being added as the value to the key. I tried a couple of different methods, and took a lot of inspiration from `buftabline` a plugin I use